README.md
# Regex Library
The library package containing the widely used regex expressions

## Install

```bash
npm install @vieolo/regex-library
```

## Usage
In the target package, the functions can be imported as followed
```JS
import {  } from '@vieolo/regex-library';
```



changelog.md
# Change Log

## 0.2.0 (2022-12-15)
- Improved the email regex

#### Breaking Changes
- `emailRegex` is now `emailRegexStandalone` and `emailRegexMix`

## 0.1.4 (2022-05-06)
- Improved the package structure

## 0.1.3 (2022-01-28)
- Added date regex for `YYYY-MM-DD`, `DD/MM/YYYY`, and `MM/DD/YYYY`

## 0.1.2 (2021-10-21)
- Added `SEPAXMLTextRegex` regex

## 0.1.1 (2021-10-07)
- Added `NetherlandsPostCode` regex

## 0.1.0 (2021-06-19)
- Initial Release

index.test.ts
import {
    emailRegex,
    hoursWithLeadingZero,
    minutesWithLeadingZero,
    time24Hours,
    time24HoursWithSeconds
} from './index';





describe("Regex Tests", () => {

    it("Matches the Time regex correctly", () => {

        // This regex does not enforce the start and the end of the string. So, if a match existing in the middle of a string, it returns true
        expect(hoursWithLeadingZero.test("00")).toBeTruthy();
        expect(hoursWithLeadingZero.test("0s0")).toBeFalsy();
        expect(hoursWithLeadingZero.test("09")).toBeTruthy();
        expect(hoursWithLeadingZero.test("09s")).toBeTruthy();
        expect(hoursWithLeadingZero.test("11")).toBeTruthy();
        expect(hoursWithLeadingZero.test("23")).toBeTruthy();
        expect(hoursWithLeadingZero.test("24")).toBeFalsy();
        expect(hoursWithLeadingZero.test("111")).toBeTruthy();
        expect(hoursWithLeadingZero.test("0")).toBeFalsy();


        // This regex does not enforce the start and the end of the string. So, if a match existing in the middle of a string, it returns true
        expect(minutesWithLeadingZero.test("00")).toBeTruthy();
        expect(minutesWithLeadingZero.test("0s0")).toBeFalsy();
        expect(minutesWithLeadingZero.test("09")).toBeTruthy();
        expect(minutesWithLeadingZero.test("09s")).toBeTruthy();
        expect(minutesWithLeadingZero.test("11")).toBeTruthy();
        expect(minutesWithLeadingZero.test("23")).toBeTruthy();
        expect(minutesWithLeadingZero.test("60")).toBeFalsy();
        expect(minutesWithLeadingZero.test("111")).toBeTruthy();
        expect(minutesWithLeadingZero.test("0")).toBeFalsy();

        
        // The 24 hours time regexes (yes, the plural of regex is regexes) enforce the start and the end of the string. So, a case in the middle of the string will not be matched
        expect(time24Hours.test("0")).toBeFalsy();
        expect(time24Hours.test("00")).toBeFalsy();
        expect(time24Hours.test("23")).toBeFalsy();
        expect(time24Hours.test("23:")).toBeFalsy();
        expect(time24Hours.test("23:1")).toBeFalsy();
        expect(time24Hours.test("23:s1")).toBeFalsy();
        expect(time24Hours.test("23:60")).toBeFalsy();
        expect(time24Hours.test("24:11")).toBeFalsy();
        expect(time24Hours.test("24s11")).toBeFalsy();
        expect(time24Hours.test("2s:11")).toBeFalsy();
        expect(time24Hours.test("s22:11")).toBeFalsy();
        expect(time24Hours.test("22:11s")).toBeFalsy();
        expect(time24Hours.test("00:11")).toBeTruthy();
        expect(time24Hours.test("23:59")).toBeTruthy();
        expect(time24Hours.test("18:42")).toBeTruthy();
        expect(time24Hours.test("04:05")).toBeTruthy();


        expect(time24HoursWithSeconds.test("0")).toBeFalsy();
        expect(time24HoursWithSeconds.test("00")).toBeFalsy();
        expect(time24HoursWithSeconds.test("23")).toBeFalsy();
        expect(time24HoursWithSeconds.test("23:")).toBeFalsy();
        expect(time24HoursWithSeconds.test("23:1")).toBeFalsy();
        expect(time24HoursWithSeconds.test("23:s1")).toBeFalsy();
        expect(time24HoursWithSeconds.test("23:60")).toBeFalsy();
        expect(time24HoursWithSeconds.test("24:11")).toBeFalsy();
        expect(time24HoursWithSeconds.test("24s11")).toBeFalsy();
        expect(time24HoursWithSeconds.test("2s:11")).toBeFalsy();
        expect(time24HoursWithSeconds.test("s22:11")).toBeFalsy();
        expect(time24HoursWithSeconds.test("22:11s")).toBeFalsy();
        expect(time24HoursWithSeconds.test("00:11")).toBeFalsy();
        expect(time24HoursWithSeconds.test("23:59")).toBeFalsy();
        expect(time24HoursWithSeconds.test("18:42")).toBeFalsy();
        expect(time24HoursWithSeconds.test("04:05")).toBeFalsy();
        expect(time24HoursWithSeconds.test("00:11s11")).toBeFalsy();
        expect(time24HoursWithSeconds.test("23:59:1s")).toBeFalsy();
        expect(time24HoursWithSeconds.test("18:42:s1")).toBeFalsy();
        expect(time24HoursWithSeconds.test("04:05:60")).toBeFalsy();
        expect(time24HoursWithSeconds.test("00:11:20")).toBeTruthy();
        expect(time24HoursWithSeconds.test("23:59:59")).toBeTruthy();
        expect(time24HoursWithSeconds.test("18:42:00")).toBeTruthy();
        expect(time24HoursWithSeconds.test("04:05:06")).toBeTruthy();

    })

});



test/index.test.ts
import {
    hoursWithLeadingZero,
    minutesWithLeadingZero,
    time24Hours,
    time24HoursWithSeconds,
    dateDDMMYYYY,
    dateMMDDYYYY,
    dateYYYYMMDD,
    NetherlandsPostCode,
    SEPAXMLTextRegex,
    emailRegexMix,
    emailRegexStandAlone,
} from '../src/index';





describe("Regex Tests", () => {

    it("Matches the Time regex correctly", () => {

        // This regex does not enforce the start and the end of the string. So, if a match existing in the middle of a string, it returns true
        expect(hoursWithLeadingZero.test("00")).toBeTruthy();
        expect(hoursWithLeadingZero.test("0s0")).toBeFalsy();
        expect(hoursWithLeadingZero.test("09")).toBeTruthy();
        expect(hoursWithLeadingZero.test("09s")).toBeTruthy();
        expect(hoursWithLeadingZero.test("11")).toBeTruthy();
        expect(hoursWithLeadingZero.test("23")).toBeTruthy();
        expect(hoursWithLeadingZero.test("24")).toBeFalsy();
        expect(hoursWithLeadingZero.test("111")).toBeTruthy();
        expect(hoursWithLeadingZero.test("0")).toBeFalsy();


        // This regex does not enforce the start and the end of the string. So, if a match existing in the middle of a string, it returns true
        expect(minutesWithLeadingZero.test("00")).toBeTruthy();
        expect(minutesWithLeadingZero.test("0s0")).toBeFalsy();
        expect(minutesWithLeadingZero.test("09")).toBeTruthy();
        expect(minutesWithLeadingZero.test("09s")).toBeTruthy();
        expect(minutesWithLeadingZero.test("11")).toBeTruthy();
        expect(minutesWithLeadingZero.test("23")).toBeTruthy();
        expect(minutesWithLeadingZero.test("60")).toBeFalsy();
        expect(minutesWithLeadingZero.test("111")).toBeTruthy();
        expect(minutesWithLeadingZero.test("0")).toBeFalsy();

        
        // The 24 hours time regexes (yes, the plural of regex is regexes) enforce the start and the end of the string. So, a case in the middle of the string will not be matched
        expect(time24Hours.test("0")).toBeFalsy();
        expect(time24Hours.test("00")).toBeFalsy();
        expect(time24Hours.test("23")).toBeFalsy();
        expect(time24Hours.test("23:")).toBeFalsy();
        expect(time24Hours.test("23:1")).toBeFalsy();
        expect(time24Hours.test("23:s1")).toBeFalsy();
        expect(time24Hours.test("23:60")).toBeFalsy();
        expect(time24Hours.test("24:11")).toBeFalsy();
        expect(time24Hours.test("24s11")).toBeFalsy();
        expect(time24Hours.test("2s:11")).toBeFalsy();
        expect(time24Hours.test("s22:11")).toBeFalsy();
        expect(time24Hours.test("22:11s")).toBeFalsy();
        expect(time24Hours.test("00:11")).toBeTruthy();
        expect(time24Hours.test("23:59")).toBeTruthy();
        expect(time24Hours.test("18:42")).toBeTruthy();
        expect(time24Hours.test("04:05")).toBeTruthy();


        expect(time24HoursWithSeconds.test("0")).toBeFalsy();
        expect(time24HoursWithSeconds.test("00")).toBeFalsy();
        expect(time24HoursWithSeconds.test("23")).toBeFalsy();
        expect(time24HoursWithSeconds.test("23:")).toBeFalsy();
        expect(time24HoursWithSeconds.test("23:1")).toBeFalsy();
        expect(time24HoursWithSeconds.test("23:s1")).toBeFalsy();
        expect(time24HoursWithSeconds.test("23:60")).toBeFalsy();
        expect(time24HoursWithSeconds.test("24:11")).toBeFalsy();
        expect(time24HoursWithSeconds.test("24s11")).toBeFalsy();
        expect(time24HoursWithSeconds.test("2s:11")).toBeFalsy();
        expect(time24HoursWithSeconds.test("s22:11")).toBeFalsy();
        expect(time24HoursWithSeconds.test("22:11s")).toBeFalsy();
        expect(time24HoursWithSeconds.test("00:11")).toBeFalsy();
        expect(time24HoursWithSeconds.test("23:59")).toBeFalsy();
        expect(time24HoursWithSeconds.test("18:42")).toBeFalsy();
        expect(time24HoursWithSeconds.test("04:05")).toBeFalsy();
        expect(time24HoursWithSeconds.test("00:11s11")).toBeFalsy();
        expect(time24HoursWithSeconds.test("23:59:1s")).toBeFalsy();
        expect(time24HoursWithSeconds.test("18:42:s1")).toBeFalsy();
        expect(time24HoursWithSeconds.test("04:05:60")).toBeFalsy();
        expect(time24HoursWithSeconds.test("00:11:20")).toBeTruthy();
        expect(time24HoursWithSeconds.test("23:59:59")).toBeTruthy();
        expect(time24HoursWithSeconds.test("18:42:00")).toBeTruthy();
        expect(time24HoursWithSeconds.test("04:05:06")).toBeTruthy();

    })


    it("Matches the Time regex correctly", () => {
        
        expect(emailRegexStandAlone.test("example@example.com")).toBeTruthy();
        expect(emailRegexStandAlone.test("Example._-1234@Exa.co.UK")).toBeTruthy();                
        expect(emailRegexStandAlone.test("Example._-1234@Exa.co.UK ssds")).toBeFalsy();                
        expect(emailRegexMix.test("sd Example._-1234@Exa.co.UK ssds")).toBeTruthy();                
        expect(emailRegexStandAlone.test("Example._-1234@Exa")).toBeFalsy();                
    })


    it("Matched the Date regex correctly", () => {

        // YYYY-MM-DD
        expect(dateYYYYMMDD.test('2020-10-10')).toBeTruthy();
        expect(dateYYYYMMDD.test('2020-12-31')).toBeTruthy();
        expect(dateYYYYMMDD.test('2020-01-01')).toBeTruthy();
        expect(dateYYYYMMDD.test('2020-1-10')).toBeFalsy();
        expect(dateYYYYMMDD.test('2020-10-1')).toBeFalsy();
        expect(dateYYYYMMDD.test('2020-13-10')).toBeFalsy();
        expect(dateYYYYMMDD.test('2020-10-32')).toBeFalsy();
        expect(dateYYYYMMDD.test('2020-10-100')).toBeFalsy();
        expect(dateYYYYMMDD.test('12020-10-10')).toBeFalsy();
        expect(dateYYYYMMDD.test('2020--10-10')).toBeFalsy();
        expect(dateYYYYMMDD.test('2020-10--10')).toBeFalsy();
        expect(dateYYYYMMDD.test('2020/10/10')).toBeFalsy();


        // DD-MM-YYYY
        expect(dateDDMMYYYY.test('10/10/2020')).toBeTruthy();
        expect(dateDDMMYYYY.test('31/12/2020')).toBeTruthy();
        expect(dateDDMMYYYY.test('01/01/2020')).toBeTruthy();
        expect(dateDDMMYYYY.test('10/1/2020')).toBeFalsy();
        expect(dateDDMMYYYY.test('1/10/2020')).toBeFalsy();
        expect(dateDDMMYYYY.test('10/13/2020')).toBeFalsy();
        expect(dateDDMMYYYY.test('32/10/2020')).toBeFalsy();
        expect(dateDDMMYYYY.test('100/10/2020')).toBeFalsy();
        expect(dateDDMMYYYY.test('10/100/2020')).toBeFalsy();
        expect(dateDDMMYYYY.test('10/10/12020')).toBeFalsy();
        expect(dateDDMMYYYY.test('10//10/2020')).toBeFalsy();
        expect(dateDDMMYYYY.test('10/10//2020')).toBeFalsy();
        expect(dateDDMMYYYY.test('10-10-2020')).toBeFalsy();


        // MM-DD-YYYY
        expect(dateMMDDYYYY.test('10/10/2020')).toBeTruthy();
        expect(dateMMDDYYYY.test('12/31/2020')).toBeTruthy();
        expect(dateMMDDYYYY.test('01/01/2020')).toBeTruthy();
        expect(dateMMDDYYYY.test('1/10/2020')).toBeFalsy();
        expect(dateMMDDYYYY.test('10/1/2020')).toBeFalsy();
        expect(dateMMDDYYYY.test('13/10/2020')).toBeFalsy();
        expect(dateMMDDYYYY.test('10/32/2020')).toBeFalsy();
        expect(dateMMDDYYYY.test('100/10/2020')).toBeFalsy()
        expect(dateMMDDYYYY.test('10/100/2020')).toBeFalsy();
        expect(dateMMDDYYYY.test('10/10/12020')).toBeFalsy();
        expect(dateMMDDYYYY.test('10//10/2020')).toBeFalsy();
        expect(dateMMDDYYYY.test('10/10//2020')).toBeFalsy();
        expect(dateMMDDYYYY.test('10-10-2020')).toBeFalsy();

    })


    it("Matches the Address regex correctly", () => {
        // This regex does not enforce the start and the end of the string. So, if a match existing in the middle of a string, it returns true
        expect(NetherlandsPostCode.test("")).toBeFalsy();
        expect(NetherlandsPostCode.test("A")).toBeFalsy();
        expect(NetherlandsPostCode.test("1")).toBeFalsy();
        expect(NetherlandsPostCode.test("10")).toBeFalsy();
        expect(NetherlandsPostCode.test("101")).toBeFalsy();
        expect(NetherlandsPostCode.test("1000")).toBeFalsy();
        expect(NetherlandsPostCode.test("1009")).toBeFalsy();
        expect(NetherlandsPostCode.test("1009 AA")).toBeFalsy();
        expect(NetherlandsPostCode.test("1009AA")).toBeFalsy();
        expect(NetherlandsPostCode.test("1010 AA")).toBeFalsy();
        expect(NetherlandsPostCode.test("1011")).toBeFalsy();
        expect(NetherlandsPostCode.test("1011 A")).toBeFalsy();
        expect(NetherlandsPostCode.test("1011 AA")).toBeTruthy();
        expect(NetherlandsPostCode.test("1020AB")).toBeTruthy();
        expect(NetherlandsPostCode.test("1099ZZ")).toBeTruthy();
        expect(NetherlandsPostCode.test("1100 EW")).toBeTruthy();
        expect(NetherlandsPostCode.test("1111 RA")).toBeTruthy();
        expect(NetherlandsPostCode.test("1999GH")).toBeTruthy();
        expect(NetherlandsPostCode.test("2999HG")).toBeTruthy();
        expect(NetherlandsPostCode.test("3000za")).toBeTruthy();
        expect(NetherlandsPostCode.test("9999Vb")).toBeTruthy();
        expect(NetherlandsPostCode.test("9999vB")).toBeTruthy();
        expect(NetherlandsPostCode.test("999vB")).toBeFalsy();
        expect(NetherlandsPostCode.test("8765  AB")).toBeFalsy();
        expect(NetherlandsPostCode.test("8765 ABZ")).toBeFalsy();
    })

    it("Matched SEPA XML text correctly", () => {
        expect(SEPAXMLTextRegex.test("")).toBeTruthy();
        expect(SEPAXMLTextRegex.test("8765 ABZ")).toBeTruthy();
        expect(SEPAXMLTextRegex.test("Note: Senders must ensure that the content of Identifiers/reference data elements is restricted to the restricted basic Latin character set (across). If non supported characters are used in these fields they may lead to rejection of files or transactions in the payment chain.")).toBeTruthy();
        expect(SEPAXMLTextRegex.test('a b c d e f g hi j k l m no p q r s t u v w x y z A B C D E F G H I J K L M N O P Q R S T U V W X Y Z 0 1 2 3 4 5 6 7 8 9 / - ? : ( ) . , + & < > "')).toBeTruthy();
        expect(SEPAXMLTextRegex.test("Vieolo OÜ")).toBeFalsy();
        expect(SEPAXMLTextRegex.test("'")).toBeFalsy();
        expect(SEPAXMLTextRegex.test("`")).toBeFalsy();
        expect(SEPAXMLTextRegex.test("_")).toBeFalsy();
        expect(SEPAXMLTextRegex.test("!")).toBeFalsy();
        expect(SEPAXMLTextRegex.test(";")).toBeFalsy();
        expect(SEPAXMLTextRegex.test("Optionally, if the personal account or organization in which you're creating uses any GitHub Apps from GitHub Marketplace, select any apps you'd like to use in the repository.")).toBeFalsy();        
    })

});



index.js
//#region Time Regex
/** Used in cases such as time expressions like 14:23 */
export let hoursWithLeadingZero = /0{0}([0-1]{1}[0-9]{1}|[2]{1}[0-3]{1})/;
/** Used in cases such as time expressions like 14:23 */
export let minutesWithLeadingZero = /0{0}[0-5]{1}[0-9]{1}/;
/** Regex matching 14:23 */
export let time24Hours = new RegExp("^" + hoursWithLeadingZero.source + "\:" + minutesWithLeadingZero.source + "$");
/** Regex matching 14:23:56 */
export let time24HoursWithSeconds = new RegExp("^" + hoursWithLeadingZero.source + "\:" + minutesWithLeadingZero.source + "\:" + minutesWithLeadingZero.source + "$");
//#endregion
//#region Email Regex
/** Regex matching example@example.com */
export let emailRegex = /[a-z0-9!#$%&'*+\/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+\/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?/gm;
//#endregion


version_bump.mjs
import { execSync } from 'child_process';
import * as fs from 'fs';

if (process.argv.length == 2) throw 'Please select add the bump level. The accepted options are "build", "minor", and "major"';

/**
 * @type {'build' | 'minor' | 'major'}
 */
let level = process.argv.slice(2);

let packageJSON = JSON.parse(fs.readFileSync('./package.json').toString());

/** @type {number[]} */
let cv = packageJSON.version.split('.').map(a => parseInt(a));

if (level == 'build') {
    cv[2] += 1;
} else if (level == 'minor') {
    cv[2] = 0;
    cv[1] += 1;
} else {
    cv[2] = 0;
    cv[1] = 0;
    cv[0] += 1;
}

// Updated Version
let fv = cv.join('.');

console.log(`---------------------------------------`);
console.log(`Bumping the version to ${fv}`);

packageJSON.version = fv;

// Writing the package.json
fs.writeFileSync('./package.json', JSON.stringify(packageJSON, null, 2));


// Updating the ReadMe file
let readMe = fs.readFileSync('./README.md').toString();
let readMeLines = readMe.split('\n');
/** @type {string[]} */
let readMeNewLines = [];

// TODO: replace the package name with the actual name from README file
for (let i = 0; i < readMeLines.length; i++) {
    const line = readMeLines[i];
    if (line.includes(`"@vieolo/regex-library": "github:Vieolo`)) {
        readMeNewLines.push(`"@vieolo/regex-library": "github:Vieolo/regex-library-js#${fv}"`)
    } else {
        readMeNewLines.push(line);
    }
}

fs.writeFileSync('./README.md', readMeNewLines.join('\n'));


// Adding lines to the Change Log
let changelog = fs.readFileSync('./changelog.md').toString();
let changeLogLines = changelog.split('\n').slice(1);

changeLogLines.unshift(...[
    "# Change Log",
    "",
    `## ${fv} (${today()})`,
    "- TODO: add the changes",
    "",
    "#### Deprecation",
    "- (Optional) deprecations in this version",
    "",
    "#### Breaking Changes",
    "- (Optional) breaking changes in this version"
])
fs.writeFileSync('./changelog.md', changeLogLines.join('\n'));



// Creating a build ************

// Setting the tsconfig's noEmit to false

console.log(`Transpiling the package`);

try {
    execSync('npm run build');   
    console.log(`Version ${fv} is ready to be published`);
    console.log(`---------------------------------------`);
} catch (error) {
    console.log(error);    
    console.log(`---------------------------------------`);
}



/**
 * @returns {string}
 */
function today() {
    let t = new Date();

    let y = t.getFullYear();
    let m = t.getMonth() + 1;
    let d = t.getDate();

    return `${y}-${m < 10 ? "0" + m : m}-${d < 10 ? "0" + d : d}`;
}

src/index.ts
//#region Time Regex

/** Used in cases such as time expressions like 14:23 */
export let hoursWithLeadingZero = /0{0}([0-1]{1}[0-9]{1}|[2]{1}[0-3]{1})/;

/** Used in cases such as time expressions like 14:23 */
export let minutesWithLeadingZero = /0{0}[0-5]{1}[0-9]{1}/;


/** Regex matching 14:23 */
export let time24Hours = new RegExp("^" + hoursWithLeadingZero.source + "\:" + minutesWithLeadingZero.source + "$");

/** Regex matching 14:23:56 */
export let time24HoursWithSeconds = new RegExp("^" + hoursWithLeadingZero.source + "\:" + minutesWithLeadingZero.source + "\:" + minutesWithLeadingZero.source + "$");


//#endregion


//#region Date Regex

/** Regex matching 2020-12-24 */
export let dateYYYYMMDD = /^([12]\d{3}-(0[1-9]|1[0-2])-(0[1-9]|[12]\d|3[01]))$/m;

/** Regex matching 24/12/2020 */
export let dateDDMMYYYY = /^((0[1-9]|[12]\d|3[01])\/(0[1-9]|1[0-2])\/[12]\d{3})$/m;

/** Regex matching 12/24/2020 */
export let dateMMDDYYYY = /^((0[1-9]|1[0-2])\/(0[1-9]|[12]\d|3[01])\/[12]\d{3})$/m;

//#endregion


//#region Email Regex

/** Regex matching example@example.com */
let emailRegexRaw = /[A-Za-z0-9!#$%&'*+\/=?^_`{|}~-]+(?:\.[A-Za-z0-9!#$%&'*+\/=?^_`{|}~-]+)*@(?:[A-Za-z0-9](?:[A-Za-z0-9]*[A-Za-z0-9])?\.)+[A-Za-z0-9](?:[A-Za-z0-9]*[A-Za-z0-9])?/

/** Regex matching an email address, anywhere in the text */
export let emailRegexMix = new RegExp(emailRegexRaw.source);

/** Regex matching an email address, with no other text allowed in the subject*/
export let emailRegexStandAlone = new RegExp("^" + emailRegexRaw.source + "$");


//#endregion



//#region Address Regex

/** 
 * Regex matching the post codes of addresses in Netherland 
 * A post code in Netherland, the last section before the name of the city, is made up of 4 numbers and 2 alphabets
 * The first two number indicate the area, the second two numbers indicate the locality, and the two alphabets indicate the street
 * Currently, the range of post codes are from 1011AA to 9999ZZ
 * There can be a single white space between the four number part and two alphabet part. Hence, both `1013 AA` nad `1013AA` are valid.
 */
export let NetherlandsPostCode = /^((101[1-9]|10[2-9][0-9]|1[1-9][0-9][0-9]|[2-9][0-9][0-9][0-9])\s?(([A-Z]|[a-z]){2}))$/m;

//#endregion



//#region SEPA regex

/**
 * Regex matching the texts allowed in the batch payment XML used for SEPA.
 * 
 * The allowed characters are:
 * 
 * a b c d e f g hi j k l m no p q r s t u v w x y z
 * A B C D E F G H I J K L M N O P Q R S T U V W X Y Z 01 2 3 4 5 6 7 8 9
 * / - ? : ( ) .,+ & < > "
 * 
 * Please note that [&, <, >, "] should be replaced with their XML representations: [&amp;, &lt;, &gt;, &quot;]
 * 
 * Please note that beside this regex, the text cannot have two consecutive `/` (//) and cannot start or end with a `/`
 */
export let SEPAXMLTextRegex = /^[A-Za-z0-9\s/?:<>,+&"().-]+$|^$/

//#endregion